### ISSUE TITLE
[ISSUE] Can't show namida queue in galaxy watch media controller 

### ISSUE THREAD

**Issue by Not-James-Bond** [namidaco/namida#351]
Not sure if it should be feature request or issue but here it is If I try to view queue in galaxy watch's media controller it shows 'Can't Show Namida Queue' 

## Expected Behavior
Queue should be visible in the controller

## Screenshots
![Screenshot_20240919_200101_mediacontroller](https://github.com/user-attachments/assets/ebbeec96-5bbf-481e-9b7c-ca96da306496)

**Comment by MSOB7YY** [comment]
oooh @FastCocobo which version u tried working and which doesnt?

**Comment by FastCocobo** [comment]
@MSOB7YY idk it was a while ago so I would say around version 2.5.6

**Comment by FastCocobo** [comment]
@MSOB7YY I just checked, it quickly flashes "this app doesn't support queue" then switches to the can't show queue one

**Comment by MSOB7YY** [comment]
i see, its definetly a namida issue and im not sure if we would be able to solve it soon. at least next/prev buttons are working for now

**Comment by FastCocobo** [comment]
Found a solution to this. It's not namida's fault. In Developer settings on android -> Networking -> Bluetooth AVRCP Version -> set it to 1.6 or higher

**Comment by FastCocobo** [comment]
> @FastCocobo i see, i dont remember doing anything related to internal queue, we dont expose the queue state to the underlying exoplayer in the first place cuz we relying on a custom one.
> maybe the android auto detection support that was added in v4.0.85 caused it? is it possible u try [v4.0.8 clone](https://github.com/namidaco/namida-snapshots/releases/tag/4.0.8-beta%2B240831200) and test it?

Nope, it worked on v4.0.8. Some android Manufactorers use a lower level of AVRCP since it means some info isn't conveyed but there is better device support. thr higher versions means that the playlist info wasn't being sent. (as for why it worked on other apps but not namida idk)

as for me saying it used to work, I'm not crazy, my old phone I set it to a higher level but when I retested it I forgot to set it high on the new phone XD

**Comment by MSOB7YY** [comment]
@Not-James-Bond  @FastCocobo 
> However, the issue now is that the queue shows 1/1, so this problem is probably due to the way Namida is handling the queue, which the watch can't interpret.

yes exactly, i wrote a queue sync thingy a while back but left it unused (cuz as i tested it showed no difference on android auto)
i will enable using it and notify u again to check

> as for me saying it used to work, I'm not crazy, my old phone I set it to a higher level but when I retested it I forgot to set it high on the new phone XD

so its fixed for u too, im glad hahgahgaha

**Comment by MSOB7YY** [comment]
@Not-James-Bond 
 ### can you check if queue shows correct count and wether its generally stable or not in [**v4.9.0**](https://github.com/namidaco/namida-snapshots) ?

**Comment by FastCocobo** [comment]
> @Not-James-Bond 
>  ### can you check if queue shows correct count and wether its generally stable or not in [**v4.9.0**](https://github.com/namidaco/namida-snapshots) ?

I just checked, it still only shows the track your playing

**Comment by MSOB7YY** [comment]
damn shi

**Comment by MSOB7YY** [comment]
it should be visible with 5.0.1
latest beta here https://github.com/namidaco/namida-snapshots/releases

pls reopen if the issue still exists in beta

**Comment by C0C0B01** [comment]
It's still an issue, it shows up but it doesn't show progress. I can skip next but it doesn't update on the watch

**Comment by MSOB7YY** [comment]
@C0C0B01 any screenshots or small video to showcase?

**Comment by C0C0B01** [comment]
> @C0C0B01 any screenshots or small video to showcase?

I can in a bit but I'm busy

**Comment by C0C0B01** [comment]
New issue: it works perfectly if your only listening to one song on loop, but as soon as it switches song it dies and there is no more control

**Comment by C0C0B01** [comment]
> @C0C0B01 can u checj if 4.9.7 works correctly or not?

uhhhh just checked and in the 5 minutes of testing it was absolutely flawless, no delay, updated correctly, synced time, could control it, it even showed the backgrounds correctly

**Comment by MSOB7YY** [comment]
@C0C0B01 oh my god what could possibly gone wrong, u confirm that the issue starts with 4.9.8 tho? (can u check ^^)

**Comment by C0C0B01** [comment]
> @C0C0B01 oh my god what could possibly gone wrong, u confirm that the issue starts with 4.9.8 tho? (can u check ^^)

Issue starts with 4.9.8 yeah

**Comment by MSOB7YY** [comment]
@C0C0B01 nice, so to make it sure, does the issue happen with local files or yt section? or both?

**Comment by C0C0B01** [comment]
@MSOB7YY yt is the same for both

**Comment by C0C0B01** [comment]
wtf just found new issue I'm listening to a song on yt, went to my playlist and both started playing at the same time and I have no control over the yt one there isn't even a miniplayer

**Comment by MSOB7YY** [comment]
@Not-James-Bond can u also check if u still have this issue on v5.1.2?

**Comment by C0C0B01** [comment]
it's working

**Comment by Not-James-Bond** [comment]
@C0C0B01 , has the queue issue been resolved for you? I installed version 5.1.2, but I'm still encountering the queue 1/1 problem. I can skip to the next song, pause, and play from my watch.

**Comment by C0C0B01** [comment]
@Not-James-Bond uhhhh just realised that I'm an idiot and completely forgot about the queue being able to be seen on the watch. and no it doesn't work lol

**Comment by theheeper55** [comment]
i just wanna see the album art in android auto bro

**Comment by MSOB7YY** [comment]
should be fixed in v6, sry for the wait ^^

**Comment by Not-James-Bond** [comment]
> should be fixed in v6, sry for the wait ^^

Finally! Thanks for fixing it. It wasn’t a big/imp issue but I’m glad it’s sorted. 🎉👍
